strawberry_django/fields/types.py

```diff
@@ -0,0 +1,159 @@
+import datetime, decimal, uuid
+import django
+import strawberry
+from django.db.models import fields
+from django.db.models.fields.reverse_related import ForeignObjectRel, ManyToOneRel
+from strawberry.arguments import UNSET
+from typing import List, Optional
+from .. import filters
+
+class auto:
+    pass
+
+@strawberry.type
+class DjangoFileType:
+    name: str
+    path: str
+    size: int
+    url: str
+
+@strawberry.type
+class DjangoImageType(DjangoFileType):
+    width: int
+    height: int
+
+@strawberry.type
+class DjangoModelType:
+    pk: strawberry.ID
+
+@strawberry.input
+class OneToOneInput:
+    set: Optional[strawberry.ID]
+
+@strawberry.input
+class OneToManyInput:
+    set: Optional[strawberry.ID]
+
+@strawberry.input
+class ManyToOneInput:
+    add: Optional[List[strawberry.ID]] = UNSET
+    remove: Optional[List[strawberry.ID]] = UNSET
+    set: Optional[List[strawberry.ID]] = UNSET
+
+@strawberry.input
+class ManyToManyInput:
+    add: Optional[List[strawberry.ID]] = UNSET
+    remove: Optional[List[strawberry.ID]] = UNSET
+    set: Optional[List[strawberry.ID]] = UNSET
+
+field_type_map = {
+    fields.AutoField: strawberry.ID,
```

Comment:
I think its important we can extend this. Maybe something like:
```py
field_type_map = {}

def register_field_type...

@strawberry.input
@register_field_type
class SomeClass
```
But we could add this later I guess

Reply:
Users can already override all these types by not using `auto` type. Only fields annotated with `auto` are using these maps.

One may want to override also scalar types like this `register_field_type(fields.AutoField, UUID)`. Would it makes sense to add this into settings?

Reply:
I wouldn't want to clog my settings with this. Especially because you might want to create custom scalars here also.

Reply:
That's good point